### DISCUSSION TITLE
Improving the Credits

### DISCUSSION THREAD

**Discussion by QuintillusCFC** [C7-Game/Prototype#311] [Ideas]
Off and on, I've been in conversations whose gist is, how do we make sure we give proper and visible credit for C7 contributions?  So far, there are three main ways we give credit - the credits in-game, call-outs of who contributed to releases at CivFanatics, and that you can always show someone your GitHub profile.

But there those have some potential shortcomings, and it would be nice to make it more convenient to be able to put C7 contributions in the "open source contributions" part of your resumé, and be able to quantify that if need be.  By now, C7 has had a significant amount of work contributed by a good number of people, and especially for those who are earlier in their careers, that could be useful and a practical benefit of the project.

The shortcomings, as I see it, are:

 - The in-game credits require someone to run the game, and thus own a copy of Civ III.  They also only indicate that you made at least one contribution at some point in the project's history.
 - The CivFanatics write-ups are decent for the "community prestige" part of recognition, but being per-release, there's no view of the contribution over time unless you look at all of them.
 - The GitHub Overview is decent, but is across all your projects, potentially including work projects or non-collaborative projects.

So the idea has been risen to try to quantify contributions in some way.  This will obviously be imperfect, and won't capture all types of contributions (e.g. discussions), but just like scoring a lot of points or getting a lot of rebounds in basketball can get a recruiter's attention, contributing a lot of code, art, reviews, etc. is a number that can get a recruiter's attention.  "Significant contributions to an open-source project in 2022" and being able to back it up with a reference is marketable.

With that in mind, I got some statistics on our Carthage (Preview 1 and Preview 2) contributions.  First up is the number of issues fixed with code contributions, of which there are 47 total, one split between two people:

Quintillus - 23.5
maxpetul - 12
pcen - 8
WildWeazel - 1.5
Sean-Brown - 1
kright - 1

Next is the number of code reviews.  I gave a point for everyone who commented on a review before it was merged.  There also were multiple pull reviews for some issues; both of these resulted in the total number being higher.

pcen - 19
Quintillus - 18
WildWeazel - 12
Sean-Brown - 5
maxpetul - 4
JimOfLeisure - 1
TinaFemea - 1

I haven't gathered statistics for Babylon or Aztec yet (and we weren't as meticulous about keeping track of our work then).  But that would increase the totals, especially for some of our early contributors such as JimOfLeisure.

Keeping track of this on a credits section of the Wiki and/or project web site, updated on each release or monthly, would make it useful as a reference, or establishing core contributors for potential JetBrains Rider licenses as an open source project.  We could also come up with a combined metric for ordering the Credits page by total contribution.  My initial hunch on a figure is Total = Issues fixed * 2 + PR reviews, on the theory that fixing an issue takes twice as much time or more than reviewing a pull request on average.  At some point this would include art as well, or perhaps we'd count art separately from development.

Thoughts?  Obviously I write a lot of code for the project, so I to be a completely disinterested party, but I'm not the only one, and pcen is a reviewer extraordinaire.

**Comment by pcen** [comment]
We can probably use the GitHub API to pull some interesting statistics, along with the ones you've mentioned (maybe you used the API to get them?) to automatically update a README / wiki page / whatever

**Comment by QuintillusCFC** [comment]
Carthage Preview 3 stats:

Code
Quintillus - 23
Kright - 4

Reviews
WildWeazel - 6
Kright - 3
Quintillus - 2
pcen - 2

A more broad code contributor base would be great!  We're also below 50% review rate.

As a side note, I started looking at updating the formatting for the credits (getting bold, italic, etc. to work) last week, but that isn't finished yet.
